### PR TITLE
Notification Center updates

### PR DESCRIPTION
**Pull request by aqtang**
New features:

- Clear all button
- Options to dismiss individual messages
- Subtitle
- Ability to open notification center when empty, with option to pass in a default empty message
- A close button

Also made some styling changes

 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

File: canvas_modules/harness/src/client/App.js [hunk 1]
```diff
@@ -1639,8 +1639,10 @@ export default class App extends React.Component {
 			{ action: "arrangeVertically", label: "Arrange Vertically", enable: true }
 		];
 
-		const notificationConfig = { action: "notification", label: "Notifications", enable: true, notificationHeader: "Notifications" };
-		const notificationConfig2 = { action: "notification", label: "Notifications", enable: true, notificationHeader: "Notifications Canvas 2" };
+		const notificationConfig = { action: "notification", label: "Notifications", notificationHeader: "Notification Center", notificationSubtitle: "subtitle status", enable: true,
+			emptyMessage: "You don't have any notifications right now.", clearAllMessage: "Clear all" };
+		const notificationConfig2 = { action: "notification", label: "Notifications", notificationHeader: "Notification Center Canvas 2", enable: true,
+			emptyMessage: "You don't have any notifications right now.", clearAllMessage: "Clear all" };
```
Comment (tomlyn) [inline canvas_modules/harness/src/client/App.js:1645]: We will need to document the additions to notificaiton config in the configs page of the Elyra-canvas wiki.

File: canvas_modules/common-canvas/src/common-canvas/canvas-controller.js [hunk 2]
```diff
@@ -505,6 +510,7 @@ export default class CanvasController {
 	//   "content": string, html, JSX Object (Optional),
 	//   "timestamp": string (Optional)
 	//   "callback": function, the callback function when a message is clicked (Required)
+	//   "key": string, a uuid given to the message once it is added to the stack
```
[the remark this answers is not in the document]
Comment (aqtang) [reply canvas_modules/common-canvas/src/common-canvas/canvas-controller.js:513]: I'll remove this extra field, I had overlooked the use of the id

File: canvas_modules/common-canvas/src/common-canvas/canvas-controller.js [hunk 3]
```diff
@@ -495,6 +495,11 @@ export default class CanvasController {
 		this.objectModel.clearNotificationMessages();
 	}
 
+	// Removes the notification message(s) with the given key(s)
+	deleteNotificationMessages(messageKeys) {
+		this.objectModel.deleteNotificationMessages(messageKeys);
+	}
+
```
[the remark this answers is not in the document]
Comment (aqtang) [reply canvas_modules/common-canvas/src/common-canvas/canvas-controller.js:502]: removed use of keys from tests and from canvas-controller

**Comment by aqtang** [comment]
making a new PR to track this and other changes

https://github.com/elyra-ai/canvas/pull/105

**Pull request closed without merging**